lib/functions_framework.rb

```diff
@@ -142,6 +143,30 @@ def http name = DEFAULT_TARGET, &block
       self
     end
 
+    ##
+    # Define a function that responds to Typed invocations over HTTP.
+    #
+    # You must provide a name for the function, and a block that implements the
+    # function. The block should take a single `Hash` argument which will be the
+    # JSON decoded request payload. It should return a `Hash` response which
+    # will be JSON encoded and written to the response.
+    #
+    # ## Example
+    #     FunctionsFramework.http "my-sum-function" do |add_request|
```

Comment:
This example needs to say `FunctionsFramework.typed`. I would also suggest we provide an example with an actual type class since that seems to be the use case we want to promote. (Maybe we can just borrow one of the protobuf types for purposes of the example.)